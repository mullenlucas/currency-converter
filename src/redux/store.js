// import { configureStore } from '@reduxjs/toolkit';

// const store = configureStore({
//   reducer: {
//     convert: convertReducer,
//   },
// });

// export default store;
