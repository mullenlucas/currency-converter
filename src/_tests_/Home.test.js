import React from 'react';
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import store from '../redux/store';
import CurrenciesPage from '../displays/currencyPage';

describe('Home page rendering test', () => {
  test('Home page snapshot rendering...', () => {
    const tree = render(
      <BrowserRouter>
        <Provider store={store}>
          <CurrenciesPage />
        </Provider>
      </BrowserRouter>,
    );
    expect(tree).toMatchSnapshot();
  });
});
