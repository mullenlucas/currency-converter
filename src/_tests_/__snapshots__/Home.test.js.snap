// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page rendering test Home page snapshot rendering... 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="body-section"
      >
        <div
          class="home-card"
        >
          <div
            class="hc-txt"
          >
            <h2>
              Real time currency converter to US dollars
            </h2>
            <h3>
              Discover and browse worldwide currencies and learn about them!
            </h3>
          </div>
          <div
            class="hc-img"
          >
            <img
              alt="illustration-exchange"
              class="illus-img"
              src="worldbills.png"
            />
          </div>
        </div>
        <div
          id="search"
        >
          <input
            placeholder="Search Currencies"
            type="search"
            value=""
          />
        </div>
        <div
          class="cards-container"
        />
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="body-section"
    >
      <div
        class="home-card"
      >
        <div
          class="hc-txt"
        >
          <h2>
            Real time currency converter to US dollars
          </h2>
          <h3>
            Discover and browse worldwide currencies and learn about them!
          </h3>
        </div>
        <div
          class="hc-img"
        >
          <img
            alt="illustration-exchange"
            class="illus-img"
            src="worldbills.png"
          />
        </div>
      </div>
      <div
        id="search"
      >
        <input
          placeholder="Search Currencies"
          type="search"
          value=""
        />
      </div>
      <div
        class="cards-container"
      />
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
